src/stepper/cn2.f90

```diff
@@ -55,8 +61,24 @@ subroutine cn2_set_diffusion(self, dt, vorch)
              endif
 
             !$omp parallel workshare
-            diss = one / (one + dfac * hdis)
+            vdiss = one / (one + dfac * vhdis)
+            !$omp end parallel workshare
+
+#ifdef ENABLE_BUOYANCY
+
+            if (buoy_visc%nnu .eq. 1) then
+                !Update diffusion operator used in time stepping:
+                dbac = dt
+            else
+                !Update hyperdiffusion operator used in time stepping:
+                dbac = bf * dt
+             endif
+
+
+            !$omp parallel workshare
+            bdiss = one / (one + dbac * bhdis)
```

Comment:
Same here as previous comment

Reply:
I can close this as this is not the time integrator using integrating factors.

Reply:
Yes, I think that is a good idea - it is only the leading order approximation to the integrating factor